app/src/main/java/com/github/houseorganizer/houseorganizer/MainScreenActivity.java

```diff
@@ -116,21 +122,39 @@ private void rotateView(View v) {
     }
 
     private void addEvent(View v) {
+        LayoutInflater inflater = LayoutInflater.from(MainScreenActivity.this);
+        final View dialogView = inflater.inflate(R.layout.event_creation, null);
+        new AlertDialog.Builder(MainScreenActivity.this)
+                .setTitle(R.string.event_creation_title)
+                .setView(dialogView)
+                .setPositiveButton(R.string.add, (dialog, id) -> pushEventAndDismiss(dialog, dialogView, v))
+                .setNegativeButton(R.string.cancel, (dialog, id) -> dialog.dismiss())
+                .show();
+    }
+
+    private void pushEventAndDismiss(DialogInterface dialog, View dialogView, View v) {
         Map<String, Object> data = new HashMap<>();
-        data.put("title", "added");
-        data.put("description", "this is the event that i added using the add button");
-        data.put("start", LocalDateTime.now().plusHours(2).toEpochSecond(ZoneOffset.UTC));
-        data.put("duration", 100);
+        final String title = ((EditText) dialogView.findViewById(R.id.new_event_title)).getText().toString();
+        final String desc = ((EditText) dialogView.findViewById(R.id.new_event_desc)).getText().toString();
+        final String date = ((EditText) dialogView.findViewById(R.id.new_event_date)).getText().toString();
+        final String duration = ((EditText) dialogView.findViewById(R.id.new_event_duration)).getText().toString();
+        Map<String, String> event = new HashMap<>();
+        event.put("title", title);
+        event.put("desc", desc);
+        event.put("date", date);
+        event.put("duration", duration);
+        if (Util.putEventStringsInData(event, data)) {
```

Comment:
How about making `Util::putEventStringsInData` return `false` in case of an error and true otherwise? This way you can change line 146 here to `if (! Util.putEventStringsInData(...))` which to me seems more intuitive.